doc/dev/static_type_checking.md

```diff
@@ -0,0 +1,1427 @@
+reading the code later may resolve it.
+
+*Globally ignore (both mypy and pyright):*
+```python
+ignore_me: int = 5  # type: ignore  https://www.github.com/Azure/azure-sdk-for-python/issues/my-issue
+```
+
+Note that it is possible to be specific in the error to ignore, instead of globally turning off type checking on
+that line. Syntax for this involves putting the specific error code in brackets of the ignore comment. Error codes
+are found next to the type checking error. 
+
+*mypy:*
+```python
+ignore_me: int = 5  # type: ignore[misc]
+```
+
+*pyright:*
+```python
+ignore_me: int = 5  # pyright: ignore[reportPrivateUsage]
+```
+
+The type checkers will continue to evaluate the line of code for other typing errors.
+
+
+More information about suppressing errors can be found in the
+official [mypy docs](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html?highlight=ignore#silencing-type-errors) and [pyright docs](https://github.com/microsoft/pyright/blob/main/docs/comments.md#line-level-diagnostic-suppression).
+
+
+### How to opt out of type checking
+
+All client libraries in the Python SDK repo are automatically opted in to running type checking. If there is a
+reason why a particular library should not run type checking, it is possible to add that library to a block list to prevent mypy/pyright
+from running checks.
+
+1) Place the package name on this block list: TODO
+2) Open an issue tracking that "library-name" should be opted in to running type checking
+
+> Note: Blocking your library from type checking is a *temporary* state. It is expected that checks are re-enabled as soon as possible.
+
+## Typing tips and guidance for the Python SDK
+
+This is not intended to be a complete guide to typing in Python. This section covers some common typing scenarios
+encountered when working on the SDK and provides some tips for you to get started. Typing is relatively new to Python,
+with over 20 PEPs and counting. Please check the [typing](https://docs.python.org/3/library/typing.html) documentation for the most up-to-date information.
+
+### Use typing.Any sparingly
+
+`typing.Any`, like the name suggests, assumes that a type can be _anything_ - it is compatible with every other type,
+and vice versa. This is what allows gradual typing in Python, as an untyped function like this...
+
+```python
+def add_things(x, y):
+    return x + y
+```
+
+is seen by the type checker as this...
+
+```python
+from typing import Any
+
+
+def add_things(x: Any, y: Any) -> Any:
+    return x + y
+```
+
+`Any` should be used sparingly since it effectively turns the type checker off. If the type can be narrowed down to
+something more specific, that is preferred since over usage of `Any` can undermine the type checker's ability to find
+bugs in other parts of the code. Additionally, `Any` does not benefit our customers who may be using type checking with
+our libraries.
```

Comment:
Do we want to provide examples where use of `Any` is ok and in line with the patterns seen in the SDK overall ?

Reply:
I tried to cover this in the variadic arguments (**kwargs) section. Wondering what other examples you're thinking of? I don't think we tend to use `Any` in public APIs (aside from kwargs) from what I've seen.

Reply:
yeah I was thinking of other than `kwargs` myself, so was wondering if you had seen any other examples used in the SDK :)